docs: add deploy and cli commands to readme (#14)
agntcy/dir: README.md

> ### Container images

Reviewer: golang packages above are not working, lets cover those as well, valid paths are:
ttps://pkg.go.dev/github.com/agntcy/dir/api
ttps://pkg.go.dev/github.com/agntcy/dir/server
ttps://pkg.go.dev/github.com/agntcy/dir/client

Author: would be nice to have a badge on the top of the README to link to these

Author: let me add those common project badges in another PR

task test:e2e
```

## Deployment

> To deploy the Directory, you can use the provided `Taskfile` commands to start the necessary services and deploy the Directory server.

Reviewer: ```suggestion
To deploy the Directory, you can use the provided `Taskfile` commands to start the necessary services and deploy the Directory server or alternatively you can deploy from a GitHub Helm chart release.
```

> ### Local OCI Provider and Directory

Reviewer: ```suggestion
### Local Deployment
```

> To start a local OCI registry server for storage and the Directory server Golang module, use the following commands:

Reviewer: ```suggestion
To start a local OCI registry server for storage and the Directory API server, use the following commands:
```

## CLI Commands

The Directory CLI provides `build`, `push`, and `pull` commands to interact with the Directory server. Below are the details on how to run each command.

To run these commands, you can either:
* Use the binary compiled with `task cli:compile` like `./bin/dirctl <command> <args>`
* Use the CLI module by navigating to the `cli` directory and running `go run cli.go <command> <args>`

### Build Command

The `build` command is used to compile and build the agent data model.

Usage:
```bash
dirctl build [options] <path-to-agent>
```

Options:
- `--name` : Name of the agent.
- `--version` : Version of the agent.
- `--artifact-url` : URL of the agent artifact.
- `--artifact-type` : Type of the agent artifact.
- `--author` : Authors to set for the agent. Overrides builder defaults. Example usage: `--author author1 --author author2`.
- `--category` : Categories to set for the agent. Overrides builder defaults. Example usage: `--category category1 --category category2`.
- `--created-at` : Agent creation time in RFC3339 format.
- `--llmanalyzer` : Enable LLMAnalyzer extension.

### Push Command

> The `push` command is used to publish the built agent data model to the store.

Reviewer: we should also specify that push expects an input to be JSON formatted data for agent data models

Usage:
```bash
dirctl push [options]
```

Options:
- `--from-file` : Read compiled data from file, reads from STDIN if empty.
- `--server-addr`: Directory Server API address (default "0.0.0.0:8888")

Example usage with read from STDIN: `dirctl build <args> | dirctl push`.

### Pull Command

> The `pull` command is used to retrieve agent data model from the store.

Reviewer: pull should also be documented that it will pull json-formatted data
